pkg/standalone/containers.go

```diff
@@ -79,15 +79,30 @@ func waitForContainerToStart(ctx context.Context, dockerClient *client.Client, c
 	// Unfortunately the Docker API's /containers/{id}/wait API (and the
 	// corresponding Client.ContainerWait method) don't allow waiting for
 	// container startup, so instead we'll take a polling approach.
-	for i := 5; i > 0; i-- {
+	for i := 10; i > 0; i-- {
 		if status, err := dockerClient.ContainerInspect(ctx, containerID); err != nil {
-			return fmt.Errorf("unable to inspect container (%s): %w", containerID[:12], err)
+			// There is a small gap between the time that a container ID and
+			// name are registered and the time that the container is actually
+			// created and shows up in container list and inspect requests:
+			//
+			// https://github.com/moby/moby/blob/de24c536b0ea208a09e0fff3fd896c453da6ef2e/daemon/container.go#L138-L156
+			//
+			// Given that multiple install operations tend to end up tightly
+			// synchronized by the preceeding pull operation and that this
+			// method is specifically designed to work around these race
+			// conditions, we'll allow 404 errors to pass silently (at least up
+			// until the polling time out - unfortunately we can't make the 404
+			// acceptance window any smaller than that because the CUDA-based
+			// containers are large and can take time to create).
+			if !strings.Contains(err.Error(), "No such container") {
+				return fmt.Errorf("unable to inspect container (%s): %w", containerID[:12], err)
+			}
 		} else if status.State.Status == container.StateRunning {
 			return nil
 		}
```

Comment:
```suggestion
		} else {
			switch status.State.Status {
			case container.StateRunning:
				return nil
			case container.StateCreated, container.StateRestarting:
				// wait for container to start
			default:
				return fmt.Errorf("container is in in unexpected state %q", status.State.Status)

			}
```
Do we want to exit early for unexpected states like removing, exited, or dead where waiting/polling is likely to be a waste of time?